main.go

```diff
@@ -108,26 +107,38 @@ func init() {
 	prometheus.MustRegister(version.NewCollector("blackbox_exporter"))
 }
 
-func main() {
+func probeRequestHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
I'd leave this inline with no changes. Splitting the logic across two functions is confusing.